packages/contracts/contracts/tcr/CivilTCR.sol

```diff
@@ -56,21 +58,23 @@ contract CivilTCR is RestrictedAddressRegistry {
   mapping(uint => Appeal) public appeals; // map challengeID to appeal
 
   /**
-  @notice Contructor Sets the addresses for token, voting, parameterizer, appellate, and fee recipient
+  @notice Init function calls AddressRegistry init then sets IGovernment
   @dev passes tokenAddr, plcrAddr, paramsAddr up to RestrictedAddressRegistry constructor
   @param tokenAddr Address of the TCR's intrinsic ERC20 token
   @param plcrAddr Address of a PLCR voting contract for the provided token
   @param paramsAddr Address of a Parameterizer contract
   @param govt IGovernment contract
   */
-  function CivilTCR(
+  function initTCR(
     address tokenAddr,
     address plcrAddr,
     address paramsAddr,
     IGovernment govt
-  ) public RestrictedAddressRegistry(tokenAddr, plcrAddr, paramsAddr)
+  ) public
   {
-    require(address(govt) != 0);
+    init(tokenAddr, plcrAddr, paramsAddr, "CivilTCR");
+    civilVoting = CivilPLCRVoting(plcrAddr);
```

Comment:
Nitpick: duplicates the same address, might desync with further changes in the code if not careful.
Using `CivilPLCRVoting(voting)` everywhere should work too?

Reply:
that would require modifying the AddressRegistry code more, which is what we're trying to avoid